### PR TITLE
Update experimental blocks in the feature-flags-and-experimental-interfaces doc

### PR DESCRIPTION
**Pull request by danieldudzic**
This PR updates the `feature-flags-and-experimental-interfaces.md` doc to reflect the current blocks that are behind the experimental flag.

It removes the following entries:
- Product Gallery Thumbnails ([PHP flag](https://github.com/woocommerce/woocommerce-blocks/blob/04af396b9aec5a915ad98188eded53e723a051d3/src/BlockTypesController.php#L234) | [webpack flag](https://github.com/woocommerce/woocommerce-blocks/blob/04af396b9aec5a915ad98188eded53e723a051d3/bin/webpack-entries.js#L57-L60)). - **It doesn't make sense to list each of the inner blocks of the Product Gallery here**
- Product Average Rating ([PHP flag](https://github.com/woocommerce/woocommerce-blocks/blob/1111e2fb9d6f5074df96a444b99e2fc00e4eb8d1/src/BlockTypesController.php#L229) | [webpack flag](https://github.com/woocommerce/woocommerce-blocks/blob/1111e2fb9d6f5074df96a444b99e2fc00e4eb8d1/bin/webpack-entries.js#L68-L70)) - **graduated**
- Product Rating Stars ([PHP flag](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/src/BlockTypesController.php#L230) | [webpack flag](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/bin/webpack-entries.js#L68-L70)) - **graduated**
- Product Rating Counter ([PHP flag](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/src/BlockTypesController.php#L229) | [webpack flag](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/bin/webpack-entries.js#L71-L73)) - **graduated**

## Testing Instructions

N/A

* [x] Do not include in the Testing Notes 
* [ ] Should be tested by the development team exclusively 

## WooCommerce Visibility

Required:

* [ ] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [x] N/A

## Checklist

Required:
* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [] This PR is assigned to a milestone.

Conditional:
* [ ] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [x] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

### CODE REVIEW

File: docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md [hunk 1]
```diff
@@ -35,10 +35,6 @@ The majority of our feature flagging is blocks, this is a list of them:
 ### Experimental flag
 
 -   Product Gallery ([PHP flag](https://github.com/woocommerce/woocommerce-blocks/blob/e3fe996251b270d45ecc73207ea4ad587c2dbc78/src/BlockTypesController.php#L232) | [webpack flag](https://github.com/woocommerce/woocommerce-blocks/blob/e3fe996251b270d45ecc73207ea4ad587c2dbc78/bin/webpack-entries.js#L50-L52C3)).
--   Product Gallery Thumbnails ([PHP flag](https://github.com/woocommerce/woocommerce-blocks/blob/04af396b9aec5a915ad98188eded53e723a051d3/src/BlockTypesController.php#L234) | [webpack flag](https://github.com/woocommerce/woocommerce-blocks/blob/04af396b9aec5a915ad98188eded53e723a051d3/bin/webpack-entries.js#L57-L60)).
```
Comment (tarhi-saad) [inline docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md:38]: @danieldudzic, the `Product Gallery Thumbnails` is still behind the experimental flag in trunk ([PHP](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/src/BlockTypesController.php#L236), [webpack](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/bin/webpack-entries.js#L66-L68)). Is this block supposed to be no longer experimental? Cc: @kmanijak

**Review by tarhi-saad — CHANGES_REQUESTED** [review]
Thank you, @danieldudzic, for this PR! I left a comment below!

**Review by kmanijak — COMMENTED** [review]
> Product Gallery Thumbnails ([PHP flag](https://github.com/woocommerce/woocommerce-blocks/blob/04af396b9aec5a915ad98188eded53e723a051d3/src/BlockTypesController.php#L234) | [webpack flag](https://github.com/woocommerce/woocommerce-blocks/blob/04af396b9aec5a915ad98188eded53e723a051d3/bin/webpack-entries.js#L57-L60)). - It doesn't make sense to list each of the inner blocks of the Product Gallery here

I think it does for clarity. Otherwise, it's all good! 👍

**Comment by danieldudzic** [comment]
> > Product Gallery Thumbnails ([PHP flag](https://github.com/woocommerce/woocommerce-blocks/blob/04af396b9aec5a915ad98188eded53e723a051d3/src/BlockTypesController.php#L234) | [webpack flag](https://github.com/woocommerce/woocommerce-blocks/blob/04af396b9aec5a915ad98188eded53e723a051d3/bin/webpack-entries.js#L57-L60)). - It doesn't make sense to list each of the inner blocks of the Product Gallery here
> 
> I think it does for clarity. Otherwise, it's all good! 👍

@gigitux what do you think?

Does it make sense listing all Product Gallery inner blocks in the feature-flags-and-experimental-interfaces doc?:
- Thumbnails
- Pager
- Next/Previous Buttons
- Large Image

**Comment by gigitux** [comment]
> > > Product Gallery Thumbnails ([PHP flag](https://github.com/woocommerce/woocommerce-blocks/blob/04af396b9aec5a915ad98188eded53e723a051d3/src/BlockTypesController.php#L234) | [webpack flag](https://github.com/woocommerce/woocommerce-blocks/blob/04af396b9aec5a915ad98188eded53e723a051d3/bin/webpack-entries.js#L57-L60)). - It doesn't make sense to list each of the inner blocks of the Product Gallery here
> > 
> > 
> > I think it does for clarity. Otherwise, it's all good! 👍
> 
> @gigitux what do you think?
> 
> Does it make sense listing all Product Gallery inner blocks in the feature-flags-and-experimental-interfaces doc?:
> 
> * Thumbnails
> * Pager
> * Next/Previous Buttons
> * Large Image

I think that it does. In this way, we reduce the chance of forgetting to remove the check when the block is production-ready!

**Review by gigitux — APPROVED** [review]
LGTM

**Review by tarhi-saad — APPROVED** [review]
Thank you for working on this PR, @danieldudzic! 🚢

**Pull request merged**